### PR TITLE
8277026: Remove blank lines remaining from snippet markup

### PR DESCRIPTION
**Pull request by pavelrappo**
This PR proposes to omit lines from the snippet source if those lines consist only of markup and perhaps whitespace. For example, if this PR is integrated, these three snippets will result in the same output:
```
    // @start region="example1"
    if (v.isPresent()) {
        System.out.println("v: " + v.get());
    } // @end


    if (v.isPresent()) { // @start region="example2"
        System.out.println("v: " + v.get());
    } // @end


    // @start region="example3" :
    if (v.isPresent()) {
        System.out.println("v: " + v.get());
    // @end :
    }
```
While this change complicates the mental model for markup, it removes otherwise unavoidable inner blank lines in properties files, for example:
```
    coffee=espresso
    # @highlight:
    tea=ice
```

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8277026](https://bugs.openjdk.java.net/browse/JDK-8277026): Remove blank lines remaining from snippet markup

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6623/head:pull/6623` \
`$ git checkout pull/6623`

Update a local copy of the PR: \
`$ git checkout pull/6623` \
`$ git pull https://git.openjdk.java.net/jdk pull/6623/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6623`

View PR using the GUI difftool: \
`$ git pr show -t 6623`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6623.diff">https://git.openjdk.java.net/jdk/pull/6623.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/snippet/Style.java [hunk 1]
```diff
@@ -37,21 +37,21 @@
...
     /**
      * A named style. Characters of this style are typically processed by
-     * wrapping into an HTML {@code SPAN} element with the {@code class}
+     * being wrapped into an HTML {@code SPAN} element with the {@code class}
      * attribute which is obtained from the provided name.
      */
     record Name(String name) implements Style { }
 
     /**
      * A marker of belonging to markup. Characters of this style are typically
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/snippet/Style.java:53]: The initial sentence is a little awkward. Can it be reposed to be clearer?

Comment (pavelrappo) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/snippet/Style.java:53]: Feel free to take a stab at it; I'm not good with words.

**Review by jonathan-gibbons — APPROVED** [review]
Nice.

I guess I was more expecting to filter output lines, rather than mark up input lines, but so far, I can't think of a problem case where they might be different.  I have a couple of comments/questions about comment text, but no issues with the code.  I look forward to using this in practice.

**Pull request closed without merging**